controller/Dockerfile

```diff
@@ -10,8 +10,23 @@ RUN go mod download
 COPY . .
 
 # Build the application
-RUN go build -o test-controller controller/main.go
+RUN CGO_ENABLED=0 GOOS=linux go build -o test-controller controller/main.go
```

Comment:
why do we need to set CGO_ENABLED and GOOS manually now?

Reply:
This is because the container that the initializer runs in is no longer the `go` Docker image since we need the test controller to run in the `docker` Docker image. What this is doing:
1) using the `go` container to build the binary
2) copying the binary to the `docker` container and using that as the base for the test controller.

Those two flags are to make sure that the Go contianer builds the right binary for the `docker` container.

Reply:
We do this because:
1) Docker containers are isolated from each other by design
2) Merging two Docker images is icky (you have to literally do filesystem copies)
3) we don't actually need the Go base once we build the binary